bindings/experimental/distrdf/python/DistRDF/Ranges.py

```diff
@@ -234,11 +229,89 @@ def get_percentage_ranges(treenames: List[str], filenames: List[str], npartition
     # thus we set it to one.
     last_tree_end_perc_tasks = [perc if perc > 0 else 1 for perc in percentages_wrt_files[1:]]
 
-    return [
-        TreeRangePerc(rangeid, tasktreenames[rangeid], taskfilenames[rangeid],
-                      first_tree_start_perc_tasks[rangeid], last_tree_end_perc_tasks[rangeid], friendinfo)
-        for rangeid in range(npartitions)
-    ]
+    if friendinfo is not None:
+        # We need to transmit the full list of treenames and filenames to each
+        # task, in order to properly align the full dataset considering friends.
+        return [
+            TreeRangePerc(rangeid, treenames, filenames, start_sample_idxs[rangeid], end_sample_idxs[rangeid],
+                          first_tree_start_perc_tasks[rangeid], last_tree_end_perc_tasks[rangeid], friendinfo)
+            for rangeid in range(npartitions)
+        ]
+    else:
+        # With the indexes created above, we can partition the lists of names of
+        # files and trees. Each task will get a number of trees dictated by the
+        # starting index (inclusive) and the ending index (exclusive)
+        tasktreenames = [treenames[s:e] for s, e in zip(start_sample_idxs, end_sample_idxs)]
+        taskfilenames = [filenames[s:e] for s, e in zip(start_sample_idxs, end_sample_idxs)]
+        # In each task, the starting file index is 0 and the ending file index
+        # is equal to the number of files assigned to that particular task
+        first_file_idxs = [0] * npartitions
+        last_file_idxs = [len(fns) for fns in taskfilenames]
```

Comment:
This list can be avoided by calling `len(taskfilenames[rangeid])` in the creation of each `TreeRangePerc` below